crates/cext/src/param.rs

```diff
@@ -0,0 +1,1120 @@
+///
+/// # Safety
+///
+/// name must be a pointer to valid C string to name of symbol
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_param_new(name: *const c_char) -> *mut Param {
+    if name.is_null() {
+        // if name is null return a new QkParam with value 0 to use this one for storage of calculation result
+        Box::into_raw(Box::new(Param::ParameterExpression(Arc::new(
+            ParameterExpression::from_f64(0.),
+        ))))
+    } else {
+        let name = unsafe { CStr::from_ptr(name).to_str().unwrap() };
+        if name.is_empty() {
+            // if the length of a name is zero return a new QkParam with value 0 to use this one for storage of calculation result
+            Box::into_raw(Box::new(Param::ParameterExpression(Arc::new(
+                ParameterExpression::from_f64(0.),
+            ))))
+        } else {
+            let symbol = Symbol::new(name, None, None);
+            let expr = ParameterExpression::from_symbol(symbol);
+            let param = Param::ParameterExpression(Arc::new(expr));
+            Box::into_raw(Box::new(param))
+        }
+    }
+}
+
+/// @ingroup QkParam
+/// Free the QkParam.
+///
+/// @param expr A pointer to the QkParam to free.
+///
+/// # Example
+///
+///     QkParam *a = qk_param_new("a");
+///     qk_param_free(a);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``expr`` is not either null or a valid pointer to a
+/// [ParameterExpression].
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_param_free(expr: *mut Param) {
+    if !expr.is_null() {
+        if !expr.is_aligned() {
+            panic!("Attempted to free a non-aligned pointer.")
+        }
+
+        // SAFETY: We have verified the pointer is non-null and aligned, so it should be
+        // readable by Box.
+        unsafe {
+            let _ = Box::from_raw(expr);
+        }
+    }
+}
+
+/// @ingroup QkParam
+/// Construct a new Param from value
+///
+/// @param value real number for a new QkParam
+///
+/// @return A pointer to the created QkParam.
+///
+/// # Example
+///
+///     QkParam *r = qk_param_from_value(2.5);
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_param_from_double(value: f64) -> *mut Param {
+    let value = Param::Float(value);
+    Box::into_raw(Box::new(value))
+}
+
+/// @ingroup QkParam
+/// Construct a new QkParam from complex number
+///
+/// @param value complex number for a new QkParam
+///
+/// @return A pointer to the created QkParam.
+///
+/// # Example
+///
+///     QkParam *c = qk_param_from_complex((1.0,1.0));
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_param_from_complex(value: Complex64) -> *mut Param {
+    let value = SymbolExpr::Value(Value::Complex(value));
+    let expr = ParameterExpression::from_symbol_expr(value);
+    let param = Param::ParameterExpression(Arc::new(expr));
+    Box::into_raw(Box::new(param))
+}
+
+// @ingroup QkParam
```

Comment:
Could we remove these comments? If #14686 is merged adding this in seems simple enough 🙂